config/simplenews.settings.json

```diff
@@ -1,32 +0,0 @@
-{
-    "_config_name": "simplenews.settings",
-    "simplenews_vid": "",
-    "simplenews_category_field": "field_simplenews_term",
-    "simplenews_use_combined": "never",
-    "simplenews_block_m_multiple": "Select the newsletter(s) to which you want to subscribe or unsubscribe.",
-    "simplenews_private_key": "FALSE",
-
-    "simplenews_last_cron": "0",
-    "simplenews_last_sent": "0",
-    "simplenews_use_cron": "FALSE",
-    "simplenews_throttle": "20",
-    "simplenews_spool_progress_expiration": "3600",
-
-    "simplenews_spool_expire": "0",
-    "simplenews_content_types": "",
-    "simplenews_sync_account": "TRUE",
-    "simplenews_debug": "FALSE",
-    "simplenews_format": "plain",
-
-    "simplenews_priority": "SIMPLENEWS_PRIORITY_NONE",
-    "simplenews_receipt": "0",
-    "simplenews_from_name": "",
-    "simplenews_send": "0",
-    "simplenews_test_address": "",
-
-    "simplenews_test_address_override": "0",
-    "simplenews_from_address": "",
-    "simplenews_source_cache": "SimplenewsSourceCacheBuild",
-    "simplenews_confirm_subscribe_page": "",
-    "simplenews_confirm_unsubscribe_page": ""
```

Comment:
A note on this port, you should create a simplenews.settings.json file to replace this one. That will provide default values for all of the settings on new installs of the module.

Reply:
@quicksketch, please see the below comment for declaring the default values I've just added. So with this regard I have two questions:

1. If the default values are set in the dynamic configuration file during the very first installation of the module, do we still need the `config/simplenews.settings.json` file?

2. We will be porting more modules in posterity, so I'd like to do it in right way. What is the recommended way of declaring the default values - through providing `config/simplenews.settings.json` file or using `config_set()` in the .install file would be deemed as adequate alternative?

Reply:
I will be busy with other staff during the day, so since the module is ready to use and for the purpose of saving time I will now proceed with the merge and will add the recommended file after the merge, 'cause I am not sure how to add a new file to the current PR.

Reply:
> What is the recommended way of declaring the default values - through providing config/simplenews.settings.json file or using config_set() in the .install file would be deemed as adequate alternative?

Default values for config are provided in a config file (e.g. `config/simplenews.settings.json`) and these are used when installing a module for the first time. Generally the only `config_set()` used in the `.install` file will be when converting D7 variables to Backdrop config, and this is used when upgrading a site from D7 to Backdrop.

Reply:
Done: https://github.com/backdrop-contrib/simplenews/blob/1.x-1.x/config/simplenews.settings.json

Reply:
> Generally the only `config_set()` used in the `.install` file will be when converting D7 variables to Backdrop config, and this is used when upgrading a site from D7 to Backdrop.

Or when setting a dynamic config value upon installation, I think. (e.g. something related to site name, content types, etc.)

Reply:
Thank you @BWPanda and @laryn for your comments, valuable especially at the stage of learning Backdrop ways.
  
> > Generally the only `config_set()` used in the `.install` file will be when converting D7 variables to Backdrop config, and this is used when upgrading a site from D7 to Backdrop.
> 
> Or when setting a dynamic config value upon installation, I think. (e.g. something related to site name, content types, etc.)

I thought so when implementing the following code in the .install file, because all the newsletters should have a sender's data by default until user sets with the desired ones:

```PHP
// Set default e-mail address for "From" and "To" fields.
  if (!empty($email = config_get('system.core', 'site_mail'))) {
    config_set('simplenews.settings', 'simplenews_from_address', $email);
    config_set('simplenews.settings', 'simplenews_test_address', $email);
  }
  elseif (!empty($email = ini_get('sendmail_from'))) {
    // This is not alway reliable as https://www.php.net/manual/en/mail.configuration.php
    // shows the default for `sendmail_from` is  NULL.
    config_set('simplenews.settings', 'simplenews_from_address', $email);
    config_set('simplenews.settings', 'simplenews_test_address', $email);
  }
  else {
    // As the last resort, there is no chance $user->init returns empty.
    $user = user_load('1');
    config_set('simplenews.settings', 'simplenews_from_address', $user->init);
  }
```